drivers/ksfilter/ks/misc.c

```diff
@@ -31,19 +30,14 @@ AllocateItem(
     IN POOL_TYPE PoolType,
     IN SIZE_T NumberOfBytes)
 {
-    PVOID Item = ExAllocatePoolWithTag(PoolType, NumberOfBytes, TAG_KS);
-    if (!Item)
-        return Item;
-
-    RtlZeroMemory(Item, NumberOfBytes);
-    return Item;
+    return ExAllocatePoolZero(PoolType, NumberOfBytes, TAG_KS);
```

Comment:
According to MS, `ExAllocatePoolZero` is Windows 7+ only routine! I've also checked that it does not exist in XP/2003 ntoskrnl. I'm just wondering why we don't mark it as such and use it as usually all XP/2003 compatible functions...
https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-exallocatepoolzero

> Minimum supported client | Requires WDK for Windows 10, version 2004. Targets **Windows 7** and later versions of the Windows operating system.

So, as result, doing this definitely will brake XP/2003 compatibility, and then the driver will fail to be correctly loaded due to missing export.
So I think it's better to leave it as-is, excepting the proper `NULL` returning in the `Item` checking. In essence, it does not change the functionality anyway.

Reply:
@oleg-dubinskiy If the build bots become green at the end, this is certainly because the code could be compiled (so the function exists somehow). Also, have you verified that ExAllocatePoolZero is an export in your Windows 7+ ntoskrnl? (Advice: grep through our code base ;) )

Reply:
@HBelusca I already saw it in our code base. And it's obvious that it's compiling correctly. I mean that we don't put it in `NTDDI_WIN7` conditional directive, when it's a Windows 7+ routine.
No, I did not check Windows 7 yet, because I don't have it installed. I just trust to what I see on MSDN and on XP/2003.
Let me some time to check it on Windows 7 as well. :slightly_smiling_face:

Reply:
Okay, seems your statement is correct. Just checked it on clear Windows 7 SP1 x86 and... `ExAllocatePoolZero` is also not exported there! :open_mouth: Then it seems to be implemented in the NDK header, same as in ROS, but why MSDN declares it as Windows 7+? :thinking: 
![win7_ntoskrnl](https://github.com/reactos/reactos/assets/26385117/af3128b1-b956-4147-a323-dc811858b041)

Reply:
> but why MSDN declares it as Windows 7+

Maybe to say that the function (which is an inline or something) has been introduced in the MS Windows 7+ WDK ?

Reply:
> Maybe to say that the function (which is an inline or something) has been introduced in the MS Windows 7+ WDK ?

Indeed... But then it did not exist in XP/2003 WDK yet.

> (NOTE: We have it in the "XDK" (not the NDK), which is a set of header templates that auto-generate our ntddk.h and wdm.h as well as other WDK and PSDK headers.)

Indeed. My mistake. :sweat_smile:

Reply:
> Indeed... But then it did not exist in XP/2003 WDK yet.

That's correct.

Reply:
See 380cd27.